dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/internal/BNFHeadersImpl.java

```diff
@@ -1027,7 +1046,8 @@ public boolean containsHeader(HeaderKeys key) {
         if (null == key) {
             throw new IllegalArgumentException("Null input provided");
         }
-        return (null != findHeader(key));
+        int ord = key.getOrdinal();
+        return (ord < HttpHeaderKeys.ORD_MAX) ? headersAdded.get(ord) : (null != findHeader(key));
```

Comment:
```suggestion
        return (ord < HttpHeaderKeys.ORD_MAX) ? headersAdded.get(ord) : (headersAdded.get(ord) && null != findHeader(key));
```
This way you don't call the findHeader it isn't in the headersAdded.  should it be `<` or `<=`?

Also you didn't put in my suggestions for improvements in findHeader to avoid looping in the normal case.

Reply:
Good suggestions, made the changes, just pushed the commit.